docs/code/CrossPlatform.md

```diff
@@ -0,0 +1,175 @@
+# Cross-Platform and Architecture design doc
+
+### Table of contents  <!-- omit in toc -->
+- [Cross-Platform and Architecture design doc](#cross-platform-and-architecture-design-doc)
+	- [1. Why cross-platform/architecture](#1-why-cross-platformarchitecture)
+	- [2. Current status](#2-current-status)
+		- [2.1 Problems](#21-problems)
+		- [2.2 Build](#22-build)
+		- [2.3 Managed Code](#23-managed-code)
+		- [2.4 Native Projects](#24-native-projects)
+		- [2.5 3rd Party Dependencies](#25-3rd-party-dependencies)
+	- [3 Possible Solutions](#3-possible-solutions)
+		- [3.1 Rewrite native components to work on other platforms](#31-rewrite-native-components-to-work-on-other-platforms)
+		- [3.2 Rewrite native components to be only managed code](#32-rewrite-native-components-to-be-only-managed-code)
+		- [3.3 Hybrid Between Finding Replacements and Software Fallbacks](#33-hybrid-between-finding-replacements-and-software-fallbacks)
+	- [My Suggestion](#my-suggestion)
+		- [Improving managed fallback through intrinsics](#improving-managed-fallback-through-intrinsics)
+		- [3rd Party Dependencies](#3rd-party-dependencies)
+		- [Helix](#helix)
+		- [Mobile Support](#mobile-support)
+		- [Support Grid](#support-grid)
+
+## 1. Why cross-platform/architecture
+ML.NET is an open-source machine learning framework which makes machine learning accessible to .NET developers with the same code that powers machine learning across many Microsoft products, including Power BI, Windows Defender, and Azure.
+
+ML.NET allows .NET developers to develop/train their own models and infuse custom machine learning into their applications using .NET.
+
+Currently, while .NET is able to run on many different platforms and architectures, ML.NET is only able to run on Windows, Mac, and Linux, either x86 or x64. This excludes many architectures such as Arm, Arm64, M1, and web assembly, places that .NET is currently able to run.
```

Comment:
I would say the most important platform is Arm64/Apple Silicon followed by WASM followed by Arm. So if something is going to be slow on Arm (after all we don't support intrinsics there), or possibly even missing, that shouldn't necessarily prevent us doing the right thing for Arm64.

Reply:
From what I have seen and tested so far, arm vs arm64 should mostly take the same work to get both working. The main differences should mostly be with the intrinsics, if we add them. That may change as I get farther into the implementation, but that seems to be the case currently.